internal/transport/transport_test.go

```diff
@@ -460,6 +459,81 @@ func setUpWithOptions(t *testing.T, port int, sc *ServerConfig, ht hType, copts
 	return server, ct.(*http2Client), cancel
 }
 
+// / noPingConn is a custom net.Conn wrapper that disables read deadlines
+// by overriding SetReadDeadline. It simulates a server that may or may
+// not respond to keepalive pings, controlled by the respondToPings flag.
+type noPingConn struct {
+	net.Conn
+	enforceReadDeadline bool
+}
+
+func (n *noPingConn) SetReadDeadline(t time.Time) error {
+	if n.enforceReadDeadline {
+		return n.Conn.SetReadDeadline(t)
+	}
+	return nil
+}
+
+// setUpConfigurablePingServer sets up a server with controllable ping responsiveness.
+// It takes a testing.T, ConnectOptions, a connection channel, and a control channel.
+// Returns a connected http2Client and a cancel function.
+// Use the control channel to toggle the server's ping responsiveness (true: responsive, false: unresponsive).
+func setUpConfigurablePingServer(t *testing.T, copts ConnectOptions, connCh chan net.Conn, controlCh chan bool) (*http2Client, func()) {
```

Comment:
I think we are still setting up a NoPingServer, and writes to `controlCh` only updates conn.ReadDeadline().  It doesnt seem like writing a `false` to `controlCh` does not flip the server from responding to pings to not responsive. In order for the server to respond to pings, we need to do something similar to [this](https://github.com/grpc/grpc-go/blob/master/internal/transport/transport_test.go#L394-L398). You might have decide whether to reuse `NewServerTransport` from [here](https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_server.go#L145). 

Let me know if this doesn't make sense or if i'm missing something.

Reply:
I'm going to check it, i'll ping you when its ready, thanks for the suggestions :)